cpp_exercise/src/lib/ipcmsgqlib.cc

```diff
@@ -0,0 +1,166 @@
+#include "ipcmsgqlib.h"
+#include "ipcexceptionlib.h"
+
+
+/***
+ * 
+ * MsgQueueSender Class
+ * 
+ * 
+ * **/
+
+MsgQueueSender::MsgQueueSender(const std::string & filename): _file_name(filename){
+        std::cout<<"**************************************************************"<<std::endl;
+        std::cout<<"****************************IPCSender: MESSAGE QUEUE PROTOCOL**************************"<<std::endl;
+        std::cout<<"**************************************************************"<<std::endl;
+        std::cout<<"**************************************************************"<<std::endl;
+
+   
+
+    int err;
+    remove(_msg_queue_name.c_str());
+        ret = mq_unlink(_msg_queue_name.c_str());
+
+    
+     memset(&_attrs, 0, sizeof _attrs);
+   
+
+    _attrs.mq_maxmsg = _msg_queue_maxnummsg;
+    _attrs.mq_msgsize = _msg_queue_msgsize;
+
+   FileHandler fd(_file_name);
+    
+    std::cout<<"          File Name: "<<_file_name<<std::endl;
+    std::cout<<"          File Size: "<<fd.getSize()<<" bytes"<<std::endl;
+        std::cout<<"          Loading the File. Please Wait..."<<std::endl;
+
+
+    
+
+
+    _msg_queue = mq_open(_msg_queue_name.c_str(), O_WRONLY | O_CREAT , S_IRWXU | S_IRWXG, &_attrs);
+     if(_msg_queue<0){
+         std::cout<<strerror(errno)<<std::endl;
+         throw IPCException("IPCSender ERROR: Cannot open the MSGQueue.");
+
+    }
+   
+ std::cout<<"          MSGQueue is Open: Trying to send the file  "<<std::endl;
+ 
+ fd.openForReading();
+
+        
+
+while (true)
+{
+    std::vector<unsigned char> buffer;
+    buffer = fd.readFile();
+  
+
+    err = mq_send(_msg_queue, (char *)&buffer[0],buffer.size(),_priority);
```

Comment:
buffer.data() is a better way.

And if you still want to do this, I recommend you to use `static_cast` instead of `(char*)`.

Reply:
I changed to buffer data. However the is an error when I want to use static_cast for the array 
 static_cast<char *>(buffer.data()
 invalid 'static_cast' from type 'unsigned char*' to type 'char*'

Reply:
I used reinterpret_cast in this case